fhir-bulkdata-webapp/src/main/java/com/ibm/fhir/bulkdata/jbatch/export/fast/ResourcePayloadReader.java

```diff
@@ -446,17 +445,23 @@ protected boolean isTxTimeExpired() {
      */
     private void uploadWhenReady() throws Exception {
         // Initiate the upload if we don't have one active
-        if (this.uploadId == null && cosClient != null) {
-            // Start a new upload
-            if (cosBucketPathPrefix != null && cosBucketPathPrefix.trim().length() > 0) {
-                this.currentObjectName = cosBucketPathPrefix + "/" + fhirResourceType + "_" + this.currentUploadNumber + ".ndjson";
-            } else {
-                this.currentObjectName = "job" + jobContext.getExecutionId() + "/" + fhirResourceType + "_" + this.currentUploadNumber + ".ndjson";
-            }
-            uploadId = BulkDataUtils.startPartUpload(cosClient, cosBucketName, this.currentObjectName);
+        if (this.uploadId == null) {
+            if (cosClient != null) {
+                // Start a new upload
+                if (cosBucketPathPrefix != null && cosBucketPathPrefix.trim().length() > 0) {
+                    this.currentObjectName = cosBucketPathPrefix + "/" + fhirResourceType + "_" + this.currentUploadNumber + ".ndjson";
+                } else {
+                    this.currentObjectName = "job" + jobContext.getExecutionId() + "/" + fhirResourceType + "_" + this.currentUploadNumber + ".ndjson";
+                }
+                uploadId = BulkDataUtils.startPartUpload(cosClient, cosBucketName, this.currentObjectName);
 
-            if (logger.isLoggable(Level.FINE)) {
-                logger.fine(logPrefix() + " Started new multi-part upload: '" + this.uploadId + "'");
+                if (logger.isLoggable(Level.FINE)) {
+                    logger.fine(logPrefix() + " Started new multi-part upload: '" + this.uploadId + "'");
+                }
+            } else if (provider instanceof AzureProvider) {
+                // Must be Azure
+                uploadId = "azure";
```

Comment:
For COS/S3 its used for multipart uploads I think (each part of the upload gets its own id), but how is it used for Azure?

Reply:
it's not, however there are multiple uploadId == null checks further  in the code